### PR TITLE
Fix erroneous VRL code examples in OP "Custom Metrics Governance" guide

### PR DESCRIPTION
**Pull request by tobz**
### What does this PR do?

As stated in the PR title, this fixes erroneous VRL snippets used in the "Custom Metrics Governance with Observability Pipelines" guide.

### Motivation

It was broken. :P

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/observability_pipelines/guide/custom-metrics-governance.md [hunk 1]
```diff
@@ -118,13 +117,12 @@ transforms:
...
-
-           # list of tags to keep
-           tags_blocklist = ["tag 1", "tag 2", "tag 3", "tag 4"]
+    source: |
+      # list of tags to keep
+      tags_blocklist = ["tag 1", "tag 2", "tag 3", "tag 4"]
 
-           # filter and drop any tags that don't match the tags in tags_to_keep
-           .tags = [filter(.tags) -> |_index, value| { !includes(tags_blocklist, value)}
+      # filter and drop any tags that don't match the tags in tags_to_keep
```
Comment (fuchsnj) [inline content/en/observability_pipelines/guide/custom-metrics-governance.md:124]: Should `tags_to_keep` here be `tags_allowlist`?

You should also change the `tags_blocklist` local above to `tags_allowlist`

File: content/en/observability_pipelines/guide/custom-metrics-governance.md [hunk 2]
```diff
@@ -84,52 +84,50 @@ transforms:
...
-           del(.tags.tag_to_drop) 
+    source: |
+      # Remove the `tag_to_drop` tag.
+      del(.tags.tag_to_drop) 
 ```
 
 ### Solution 2: Define an allowlist array of tags to keep
 
-In the scenario where you have many tags you want to drop and know which tags you want to keep, use the `filter()` function to define an allowlist of the tags to keep. This automatically drops any tags that aren't included in the allowlist. See the following configuration example:
+In the scenario where you want to drop all tags except for a list of allowed tags, define the list and use the `filter()` function. This automatically drops any tags that aren't included in the allowlist. See the following configuration example:
```
Comment (maycmlee) [inline content/en/observability_pipelines/guide/custom-metrics-governance.md:94]:
```suggestion
In the scenario where you want to drop all tags except for a list of allowed tags, define the allowlist and use the `filter()` function. This automatically drops any tags that aren't included in the allowlist. See the following configuration example:
```

File: content/en/observability_pipelines/guide/custom-metrics-governance.md [hunk 3]
```diff
@@ -84,52 +84,50 @@ transforms:
...
-           # filter and drop any tags that don't match the tags in tags_to_keep
-           .tags = [filter(.tags) -> |_index, value| { includes(tags_allowlist, value)}
+      # Filter and drop any tags that are not in `tags_allowlist`.
+      .tags = filter(object!(.tags)) -> |key, _value| { includes(tags_allowlist, key) }
 ```
 
 ### Solution 3: Define a blocklist array of tags to drop
 
-In the case where you know which tags you want to drop, use the `filter()` function to define a blocklist of those tags. This is done by adding `!` to `includes(tags_blocklist, value)` from Solution 2.
+In the case where you know which tags you want to drop, you can use the same steps as in Solution 2, but instead, negate the result of the `includes` call by prefixing it with `!`:
```
Comment (maycmlee) [inline content/en/observability_pipelines/guide/custom-metrics-governance.md:112]:
```suggestion
In the case where you know which tags you want to drop, you can use the same steps as in Solution 2, but negate the result of the `includes` call by prefixing it with `!`:
```

**Review by maycmlee — COMMENTED** [review]
A couple of small edits!

**Comment by tobz** [comment]
Looks like the two errors in the failed CI check are on unrelated files after merging in `master`. 😞

**Comment by maycmlee** [comment]
Thanks for merging in master! The vale linter can be a bit flakey and sometimes our build as well. Looks like the build passed this time though 🎉, otherwise I'd have re-ran it.

**Pull request merged**